### ISSUE TITLE
undefined reference to `lua_rotate'?

### ISSUE THREAD

**Issue by emacsomancer** [Elv13/awesome-configs#34]
I've been trying to compile this ('make') and I get different error messages depending on whether I have lua5.1 and lua5.2 installed alongside of lua5.3 and luajit.  With all of these installed I get a bunch of messages like:

     undefined reference to `lua_rotate'

Without lua51 and/or lua52 I get different errors.

Any tips on how to proceed?

**Comment by Elv13** [comment]
Hello,

Why are you trying to compile this yourself and why do you have some many Lua versions? Lua_rotate errors indicate a mismatch between the Lua header and binary. You can fix that by setting the environment correctly. Beside, doesn't your distribution provides Awesome 4.0 packages?

Finally, this repository requires some patch. If you want to use a simpler version, this fork https://github.com/AxelTB/awesome-configs have monkey patched the missing modules.

**Comment by emacsomancer** [comment]
Hello,
I don't know why I have so many Lua versions - Arch lets me, and some application must want to use one of the older ones.

I was trying to get your awesomewm config that looks like this - https://i.imgur.com/oUunJp9.png . 
would the fork at https://github.com/AxelTB/awesome-configs accomplish this if I'm using the ordinary Awesome 4.0 package?

**Comment by Elv13** [comment]
> would the fork at https://github.com/AxelTB/awesome-configs accomplish this if I'm using the ordinary Awesome 4.0 package?

Yes. I am one of Awesome core devs and I stopped trying to keep my config compatible with the releases. However the maintainers of this fork do. The config has many themes. This one is called "SciFi", it's currently the default in that fork.
